addon/models/occurrence-proxy.js

```diff
@@ -16,10 +16,19 @@ var OccurrenceProxy = Ember.Object.extend(Ember.Copyable, {
     );
   }),
 
-  day: Ember.computed('startingTime', 'calendar', function() {
+  day: Ember.computed('startingTime', 'calendar', 'calendar.{startingTime,startFromDate}', function() {
+    let currentDay = this.get('startingTime');
+    let firstDay;
+
+    if (this.get('calendar.startFromDate')) {
+      firstDay = this.get('calendar.startingTime');
+    } else {
+      firstDay = this.get('calendar.startingTime').startOf('isoWeek');
+    }
+
     return Day.create({
       calendar: this.get('calendar'),
-      offset: this.get('startingTime').isoWeekday() - 1
+      offset: currentDay.dayOfYear() - firstDay.dayOfYear()
```

Comment:
Just for reference, this could also be done like the following (which IMHO is more readable):

`currentDay.diff(firstDay, 'days')` _or something like that_